allocations/views.py

```diff
@@ -55,9 +56,22 @@ def get_all_alloc(request):
         json: dumps all data as serialized json.
     """
     try:
+        keycloak_client = KeycloakClient()
+        user_attributes = keycloak_client.get_all_users_attributes()
         mapper = ProjectAllocationMapper(request)
         resp = mapper.get_all_projects()
         logger.debug("Total projects: %s", len(resp))
+        for r in resp:
+            pi_attributes = user_attributes.get(r["pi"]["username"])
+            if pi_attributes:
+                institution = pi_attributes.get("affiliationInstitution")
+                country = pi_attributes.get("country")
+                r["pi"]["institution"] = (
+                    institution[0] if institution and len(institution) > 0 else None
```

Comment:
another way to do this is to change the list to an iterator and call `next`. It's a little bit clunky still, I wish python had a way to .peek() on a list and have it return None if there was no next item.

```python
r["pi"]["institution"] = next(iter(instititution), None)
```